File: backend/src/main/java/com/example/pixelplay/chess/mechanics/WhitePawnMechanics.java
package com.example.pixelplay.chess.mechanics;

import com.example.pixelplay.chess.Position;

import java.util.ArrayList;
import java.util.List;

public class WhitePawnMechanics extends PawnMechanics {

    public WhitePawnMechanics(Position position, byte square) {
        super(position, square);
    }

    @Override
    public List<Byte> attackingSquares() {
        List<Byte> squares = new ArrayList<>();

        if(!isOnLeftEdge()) {
            squares.add((byte) (square + 7));
        }
        if(!isOnRightEdge()) {
            squares.add((byte) (square + 9));
        }

        return squares;
    }


}

File: backend/src/main/java/com/example/pixelplay/chess/mechanics/WhitePawnMechanics.kt
package com.example.pixelplay.chess.mechanics

import com.example.pixelplay.chess.Position
import com.example.pixelplay.chess.PositionUtil

class WhitePawnMechanics() : PawnMechanics() {
    override val forwardIndex: Byte
        get() = 8
    override val leftDiagonalIndex: Byte
        get() = 7
    override val rightDiagonalIndex: Byte
        get() = 9
    override val startingRank: Byte
        get() = 1
}
